docopt_value.h

```diff
@@ -257,6 +257,13 @@ namespace docopt {
 	inline
 	long value::asLong() const
 	{
+		// Attempt to convert a string to a long
+		if (kind == Kind::String) {
+			// Doesn't guard against trailing characters,
+			// but doing so (if desired) would be trivial by checking pos.
+			std::size_t pos;
+			return stol(variant.strValue, &pos); // Throws if it can't convert
```

Comment:
Should we throw here if "pos != strValue.size()"? I'm not sure I like that "--foo 42bar" would give "42" and silently discard the junk at the end.